### ISSUE TITLE
Streaming upload does not work

### ISSUE THREAD

**Issue by prokopst** [requests/requests-ntlm#88]
It seems that streaming upload does not work, most likely because the file is reset [only once](https://github.com/requests/requests-ntlm/blob/master/requests_ntlm/requests_ntlm.py#L55). This is related to #87, because the data is sent three times.

For example code similar to this snippet does not work, it's always stuck:

    session = Session(auth=HttpNtlmAuth(username, password))
    with open("blabla.txt") as file:
        session.put(url="https://webdav-server/", data=file, timeout=10)

pip freeze says:

    requests==2.11.1
    requests-ntlm==1.0.0

**Comment by Lukasa** [comment]
Yeah, this looks like a bug. I think we'd need to use some of the code in the newer versions of Requests for seeking back the body if possible.

I'm on holiday for the rest of the year which means that I won't be able to get to this until next year, but I am happy to review patches if anyone feels like tackling this.

**Comment by nateprewitt** [comment]
It looks like this would be a pretty simple fix with `rewind_body` from 2.12.4. Are we alright requiring the latest version of requests being the minimum version for this module? If so, I can look into integrating it.

**Comment by ofek** [comment]
Has this been fixed?

**Comment by kamikaze** [comment]
it seems that this project is abandonned
